frontend/src/components/loginPage/loginPage.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {useState, useEffect} from 'react';
+import styled from 'styled-components';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+import { RouteComponentProps } from 'react-router-dom';
+
+import { authTypes, authActions } from '../../types/authTypes';
+import { AppState } from '../../reducers';
+import LoginBackground from '../loginBackground';
+import LoginForm from '../loginForm/loginForm';
+
+const Wrapper = styled.div`
+    height: 100%;
+`;
+
+const LoginWrapper = styled.section`
+    padding: 0 2.5rem;
+    position: relative;
+    z-index: 10;
+`;
+
+const Header = styled.h1`
+    color: rgb(249, 229, 30);
+    text-transform: uppercase;
+    text-align: center;
+    font-size: 2.4rem;
+    font-weight: bold;
+`;
+
+
+const Message = styled.div<{active: boolean, error: boolean}>` 
+    display: ${(props) => props.active ? "flex" : "None"};
+    flex-direction: row;
+    color: ${(props) => props.error ? 'rgb(249, 229, 30)' : "#99e6ff"};
+    width: 100%;
+    border-bottom: 2px solid ${(props) => props.error ? 'rgb(249, 229, 30)' : "#99e6ff"};
+    padding: 0;
+    margin-bottom: 1.5rem;
+    box-sizing: border-box;
+    align-items: center;
+
+`;
+
+const ErrorMessageIcon = styled.div<{visible: boolean}>`
+    font-size: 2rem;
+    padding-right: 0.5rem;
+    display: ${(props) => props.visible ? "block" : "None"};
+`;
+
+const MessageText = styled.p`
+
+`;
+
+interface LoginProps extends RouteComponentProps {
+    auth: (challenge: string, id: string) => void,
+    linkRequest: (phone: string) => void,
+    authError: boolean,
+    loggingIn: boolean
+}
+
+const LoginPage: React.FC<LoginProps> = (props) => {  
+    const [message, setMessage] = useState('');
+    const [error, setError] = useState(false);
+
+    useEffect(() => {
+        const query = new URLSearchParams(props.location.search);
+        const challenge = query.get('challenge')
+        const id = query.get('id')
+        if (challenge && id) {
+            props.auth(challenge, id);
+        }
+    }, [])
```

Comment:
Should we not be passing [props] here?

Reply:
no, as we use this effect only when component is mounted, not re-rendered. If I understood the question correctly

Reply:
Okay.